lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/MavenProjectCacheTest.java

```diff
@@ -76,6 +88,63 @@ public void testParentChangeReflectedToChild()
 		}
 	}
 
+	@Test
+	public void testAddFolders_didChangeWorkspaceFolders() throws Exception {
+		XMLLanguageService languageService = new XMLLanguageService();
+		IWorkspaceServiceParticipant workspaceService = languageService.getWorkspaceServiceParticipants().stream().filter(MavenWorkspaceService.class::isInstance).findAny().get();
+		assertNotNull(workspaceService);
+		
+		URI folderUri = getClass().getResource("/modules").toURI();
+		WorkspaceFolder wsFolder = new WorkspaceFolder(folderUri.toString());
+
+		// Test add folders
+		workspaceService.didChangeWorkspaceFolders(
```

Comment:
I would write `languageService.getWorkspaceServiceParticipants().forech(p -> p.didChangeWorkspaceFolders`

Reply:
We're going to test one certain participant - `MavenWorkspaceService` here in order to invoke its `didChangeWorkspaceFolders()` API .method
IMHO, Including more participants into the processing potentially may create unexpected effects causing test execution problems, at the same time not adding any useful functionality and not being a subject to test.